curve.go

```diff
@@ -115,6 +115,16 @@ func (sc StarkCurve) Add(x1, y1, x2, y2 *big.Int) (x, y *big.Int) {
 	y = y.Sub(y, y1)
 	y = y.Mod(y, sc.P)
 
+	// As elliptic curves form a group, there is an additive identity that is the equivalent of 0
+	// If 𝑃=0 or 𝑄=0, then 𝑃+𝑄=𝑄 or 𝑃+𝑄=𝑃, respectively
+	// NOTICE: to improve constant-time performance we avoid fast no-op calls by checking this at the end
+	if len(x1.Bits()) == 0 && len(y1.Bits()) == 0 {
+		return x2, y2
+	}
+	if len(x2.Bits()) == 0 && len(y2.Bits()) == 0 {
+		return x1, y1
+	}
```

Comment:
Doesn't this make it less constant time by adding more conditionals? If the calculation is already done by this point what performance do we actually gain here instead of simply returning `x, y`?

Reply:
I've updated this section and added the scalar rewrite trick from here: https://www.shiftleft.org/papers/ladder/ladder-tches.pdf